server/Arcadia.Assistant.Calendar/CalendarEventsStorageBase.cs

```diff
@@ -212,9 +216,11 @@ private void OnCalendarEventNextApproverReceived(string eventId, string nextAppr
                     oldEvent.EmployeeId
                 );
 
-                this.UpdateCalendarEvent(oldEvent, newEvent, ev =>
+                // ToDo: Here we need last approver to set event UserId
```

Comment:
Here is the place to pay attention to. This is why I needed those changes that you didn't like with `LastOrDefault`.

Reply:
you don't actually need that. Why do you need that? I mean it's ok to pass null as it was not updated actually.

Reply:
But it was updated, indirectly, but was. If someday we have field like "UpdatedBy" in CSP DB - it will fail then.

Reply:
if so eager to use that, why don't you take one from dictionary?

Reply:
Because I need your answer. Should the last approver be set as updater of automatically updated calendar event or not?

Reply:
if that's as easy as this.Approvals[eventId].Approvals.OrderBy(x => x.Timestamp).Select(x => x.UserId).LastOrDefault(), let's do that.
Point is that we don't really need to do that now, but I don't see how it hurts.

Reply:
Fixed.